taxonomy-editor-frontend/src/components/ResponsiveAppBar.tsx

```diff
@@ -85,7 +100,7 @@ export const ResponsiveAppBar = () => {
                 page.url ? (
                   <MenuItem
                     key={page.translationKey}
-                    onClick={handleCloseNavMenu}
+                    onClick={() => handleNavigation(`/${page.url}`)} // Use handleNavigation
```

Comment:
I think the method of using BrowserRouter getUserConfirmation proposed in [this article](https://dev.to/eons/detect-page-refresh-tab-close-and-route-change-with-react-router-v5-3pd) might be better (because more central). Did you try it ?